networking/p2p/src/main/java/tech/pegasys/artemis/networking/p2p/libp2p/discovery/DiscoveryService.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.networking.p2p.libp2p.discovery;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Stream;
+
+public interface DiscoveryService {
```

Comment:
Should we use the generic `Service` interface?:
```suggestion
public interface DiscoveryService implements Service {
```

Reply:
`Service` is an abstract class so the interface can't extend it, but we can use it for the implementation which makes a lot of sense.